### PR TITLE
DDF-4673 force footer to bottom of card

### PR DESCRIPTION
**Pull request by zta6**
#### What does this PR do?
The generic card view's footer was not actually being forced to the bottom of the card as an actual footer would be, but it was never a problem before because all cards that used this view filled the card with content which pushed the footer down. 

This PR forces the footer to the bottom of the card even when the card is not entirely filled with content. 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@bdeining

#### How should this be tested?

Test in downstream project

#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4673 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by samuelechu** [comment]
could you provide a before/after screenshot?

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx [hunk 1]
```diff
@@ -45,8 +46,9 @@ const Details = styled.div`
 const Footer = styled.div`
   height: ${props => props.theme.minimumButtonSize};
   text-align: right;
-  display: flex;
-  justify-content: flex-end;
+  position: absolute;
```
Comment (samuelechu) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx]: does this work fine with different zoom levels and spacing?

**Comment by zta6** [comment]
@samuelechu I'll grab some screenshots and make sure this works with various spacing/zoom later today.

**Comment by zta6** [comment]
@samuelechu Here's a before and after, respectively. 
I tried passing in style to the component from downstream but because things get nested all over the place I couldn't hit the component I needed to hit. Which is why this PR is necessary. 
ellipsisBefore
ellipsisAfter

Also tried with different spacing and zoom percentage and it all looks good to me.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx (hunk 1 above)
Comment (adimka) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx]: We might be better off doing this via flexbox and `space-between` on the root element instead.

Comment (zta6) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx]: Refactored to flexbox!

**Comment by bdeining** [comment]
Should we verify that Chrome and Firefox produce the same look and feel using flex?

**Comment by zta6** [comment]
> Should we verify that Chrome and Firefox produce the same look and feel using flex?

@bdeining tested on Firefox, tried different themes, zoom/spacing/etc. LGTM. Previously tested on Chrome.

**Comment by zta6** [comment]
build now

**Pull request closed without merging**